graph-query-ir/src/main/java/oracle/pgql/lang/spatial/Point2D.java

```
@@ -40,6 +41,35 @@ public Point2D(double x, double y, double m) {
     this.m = m;
   }
 
+  /**
+   * Create a Point2D object from a well-known text representation
+   * @param wktPoint well-known text representation of a point value
+   * @return Point2D object
+   */
+  public static Point2D fromWkt(String wktPoint) {
+    String prefix = wktPoint.substring(0, wktPoint.indexOf('('));
+    String values = wktPoint.substring(wktPoint.indexOf('(') + 1, wktPoint.indexOf(')'));
+
+    int xValEndPos = values.indexOf(' ');
+    Double xVal = Double
+        .parseDouble(values.substring(0, xValEndPos));
+
+    int yValStartPos = values.indexOf(' ') + 1;
+
+    switch (prefix) {
+      case "POINT":
+        Double yVal = Double.parseDouble(values.substring(yValStartPos, values.length()));
+        return new Point2D(xVal, yVal);
+      case "POINT M":
+        int yValEndPos = StringUtils.ordinalIndexOf(values, " ", 2);
+        yVal = Double.parseDouble(values.substring(yValStartPos, yValEndPos));
+        Double mVal = Double.parseDouble(values.substring(yValEndPos + 1, values.length()));
+        return new Point2D(xVal, yVal, mVal);
+      default:
+        return new Point2D(Double.NaN, Double.NaN);
```

here you probably want to throw an exception too.